### PR TITLE
[ Xedra Evolved ] Gracken Stomach Mutations

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
Mods "[XE] Gracken Stomach Traits" 

#### Purpose of change
Give Grackens the adaptable stomach traits that they need.  Three directions to start. Herbivore, Carnivore, Omnivore.  Herbivore can go to Ruminant.  Carnivore line probably then get eater of dead or some other carnivore evolution.  Omnivore, one option it gets will be a stomach based on Ants.  Where half the calories get turned into vitamins which can be turned into a slightly negative morale food stuff.

#### Describe the solution
Make the traits, make the harvests, make the overrides, make the items.  Make the recipes and trait consumeables.

#### Testing
Needs recipes and consumeables to give the traits.

### CODE REVIEW

File: data/mods/Xedra_Evolved/monsters/monster_overrides.json [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+  },
+  {
+    "id": "mon_ant_acid",
+    "type": "MONSTER",
+    "name": { "str": "giant acidic ant" },
+    "description": "A monstrous brown ant with a swollen abdomen that ends with a small orifice at the tip.  Glistening liquid seems to drip out periodically.",
+    "copy-from": "mon_ant_acid",
+    "harvest": "giant_acid_ant"
+  }
+]
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/monster_overrides.json:232]:
```suggestion
]

```

File: data/mods/Xedra_Evolved/mutations/xe_playable_gracken.json [hunk 2]
```diff
@@ -236,6 +253,47 @@
...
+    "mixed_effect": true,
+    "points": 1,
+    "category": [ "GRACKEN" ],
+    "threshreq": [ "THRESH_SPECIES_GRACKEN" ]
+  },
+  {
+    "type": "mutation",
+    "id": "SHADE_CARNIVORE",
+    "name": { "str": "Carnivorous Stomach" },
+    "description": "You eat an exclusively meat based diet as befits an apex predator of this planet.",
```
Comment (Night-Pryanik) [inline data/mods/Xedra_Evolved/mutations/xe_playable_gracken.json:273]:
```suggestion
    "description": "You eat an exclusively meat-based diet as befits an apex predator of this planet.",
```

File: data/mods/Xedra_Evolved/mutations/xe_playable_gracken.json [hunk 3]
```diff
@@ -236,6 +253,47 @@
...
+    "points": 1,
+    "category": [ "GRACKEN" ],
+    "threshreq": [ "THRESH_SPECIES_GRACKEN" ],
+    "flags": [ "CARNIVORE_DIET" ],
+    "vitamin_rates": [ [ "vitC", -1200 ] ]
+  },
+  {
+    "type": "mutation",
+    "id": "SHADE_HERBIVORE",
+    "name": { "str": "Carnivorous Stomach" },
```
Comment (Night-Pryanik) [inline data/mods/Xedra_Evolved/mutations/xe_playable_gracken.json:287]:
```suggestion
    "name": { "str": "Herbivore Stomach" },
```

**Comment by Maleclypse** [comment]
@Night-Pryanik Do you have any recomendations on how to fix the c++?    Or how to do this goal? ` Where half the calories get turned into vitamins which can be turned into a slightly negative morale food stuff.`  If not I can pull the ant stomach for now.

**Comment by Night-Pryanik** [comment]
What do you mean by "fix the c++"? The only thing I see that now need fixing is to sort your newly added `static const json_character_flag json_flag_CARNIVORE_DIET( "CARNIVORE_DIET" );`

**Comment by Maleclypse** [comment]
> What do you mean by "fix the c++"? The only thing I see that now need fixing is to sort your newly added `static const json_character_flag json_flag_CARNIVORE_DIET( "CARNIVORE_DIET" );`

Sorry about that I hadn't pushed the mess I have locally over here yet.   I'm going to break it out into it's own thing so I have less to look at and see if that helps me get it in a good shape.  Thanks Night!

File: src/item.cpp [hunk 4]
```diff
@@ -2724,7 +2724,7 @@ void item::food_info( const item *food_item, std::vector<iteminfo> &info,
                            std::abs( static_cast<int>( food_item->count() ) * batch ) );
     }
     if( food_item->corpse != nullptr && parts->test( iteminfo_parts::FOOD_SMELL ) &&
-        ( debug || ( g != nullptr && player_character.has_trait( trait_CARNIVORE ) ) ) ) {
+        ( debug || ( g != nullptr && player_character.has_flag( flag_CARNIVORE_DIET ) ) ) ) {
```
Comment (Maleclypse) [inline src/item.cpp:2727]:
```suggestion
        ( debug || ( g != nullptr && player_character.has_flag( json_flag_CARNIVORE_DIET ) ) ) ) {
```

File: data/mods/Xedra_Evolved/monsters/monster_overrides.json [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+    "stomach_size": 3000,
+    "vision_night": 10,
+    "path_settings": { "max_dist": 10 },
+    "anger_triggers": [ "HURT", "PLAYER_NEAR_BABY" ],
+    "fear_triggers": [ "SOUND" ],
+    "zombify_into": "mon_zombear",
+    "harvest": "bear_with_skull",
+    "dissect": "dissect_ursine_sample_small",
+    "families": [ "prof_gross_anatomy", "prof_intro_biology", "prof_physiology" ],
+    "reproduction": { "baby_monster": "mon_bear_cub", "baby_count": 1, "baby_timer": 700 },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/monster_overrides.json:59]:
```suggestion
    "reproduction": { "baby_type": { "baby_monster": "mon_bear_cub" }, "baby_count": 1, "baby_timer": 700 },
```

File: data/mods/Xedra_Evolved/monsters/monster_overrides.json [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+    "melee_skill": 5,
+    "melee_dice": 2,
+    "melee_dice_sides": 8,
+    "melee_damage": [ { "damage_type": "cut", "amount": 5 } ],
+    "dodge": 2,
+    "stomach_size": 1700,
+    "harvest": "boar_with_skull",
+    "weakpoint_sets": [ "wps_animal_quadruped" ],
+    "families": [ "prof_gross_anatomy", "prof_intro_biology", "prof_physiology", "prof_wp_ungulate" ],
+    "reproduction": { "baby_monster": "mon_boar_wild_piglet", "baby_count": 8, "baby_timer": 154 },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/monster_overrides.json:163]:
```suggestion
    "reproduction": { "baby_type": { "baby_monster": "mon_boar_wild_piglet" }, "baby_count": 8, "baby_timer": 154 },
```

File: data/mods/Xedra_Evolved/body_parts.json [hunk 7]
```diff
@@ -135,6 +135,49 @@
...
+    "heading": "s. eyes",
+    "heading_multiple": "s. eyes",
+    "encumbrance_text": "Ranged combat is hampered.",
+    "encumbrance_limit": 60,
+    "main_part": "head",
+    "opposite_part": "gracken_eyes",
+    "hit_size": 0.5,
+    "hit_difficulty": 1.15,
+    "limb_type": "sensor",
+    "limb_score_mods": [ { "limb_score": "vision", "modifier": 1 }, { "limb_score": "night_vis", "modifier": 1.5 } ],
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/body_parts.json:152]:
```suggestion
    "limb_scores": [ [ "vision", 1.0 ], [ "night_vis", 1.5 ], [ "reaction", 1 ] ],
```

File: data/mods/Xedra_Evolved/items/carnivore.json [hunk 8]
```diff
@@ -126,5 +126,58 @@
...
+    "name": { "str": "herbivore stomach" },
+    "description": "The stomach of a herbivore.  It is surprisingly durable."
+  },
+  {
+    "id": "carnivore_stomach",
+    "copy-from": "stomach",
+    "type": "COMESTIBLE",
+    "name": { "str": "carnivore stomach" },
+    "description": "The stomach of an carnivorous creature.  It is surprisingly durable."
+  },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/carnivore.json:158]:
```suggestion
  },
    {
    "id": "small_carnivore_stomach",
    "copy-from": "stomach",
    "type": "COMESTIBLE",
    "name": { "str": "small carnivore stomach" },
    "description": "The stomach of a small carnivorous creature.  It is surprisingly durable.",
    "proportional": { "weight": 0.5, "volume": 0.5, "price": 0.5, "calories": 0.5 }
  },
```

**Pull request merged**